messaging/nats.go

```diff
@@ -43,14 +43,19 @@ func ConfigureNatsConnection(config *nconf.NatsConfig, log logrus.FieldLogger) (
 }
 
 func ConnectToNats(config *nconf.NatsConfig, opts ...nats.Option) (*nats.Conn, error) {
-	if config.TLS != nil {
+	if config.Auth.Method == nconf.NatsAuthMethodUser {
```

Comment:
personally, I'd use a switch
```
switch strings.ToLower(config.Auth.Method) {
case "": 
....
}
```

Reply:
The 3rd if is a bit more complex, that's why I didn't use a switch/case.

I refactored the TLS config and now it can use a switch